app/src/main/java/io/btrshop/products/ProductsPresenter.java

```diff
@@ -29,37 +32,64 @@ public ProductsPresenter(Retrofit retrofit, ProductsContract.View mView) {
     }
 
     @Override
+
     public void scanProduct() {
         mProductsView.showScan();
     }
 
     @Override
-    public void getProduct(String ean) {
+    public void postProduct(String ean, List<BeaconJson> listBeacon) {
 
         checkNotNull(ean, "ean cannot be null!");
-        retrofit.create(ProductsService.class).getProduct(ean)
-                .subscribeOn(Schedulers.io())
-                .observeOn(AndroidSchedulers.mainThread())
-                .unsubscribeOn(Schedulers.io())
-                .subscribe(new Observer<Product>() {
-
-                    @Override
-                    public void onCompleted() {
-
-                    }
-
-                    @Override
-                    public void onError(Throwable e) {
-                        mProductsView.showError();
-                    }
-
-                    @Override
-                    public void onNext(Product product) {
-                        mProductsView.showProduct(product);
-                    }
 
-                });
 
+        if(listBeacon != null) {
```

Comment:
beacons.getListBeacons() should just return an empty list so you won't have to use this kind of condition.